backend/associations/views/association.py

```diff
@@ -133,6 +134,25 @@ def get_serializer_class(self):
         else:
             return AssociationSerializer
 
+    def list(self, request, *args, **kwargs):
+        print("listing assoc")
+        queryset = self.filter_queryset(self.get_queryset())
+        user = User.objects.get(pk=request.user.id)
+
+        queryset = [
+            a
+            for a in queryset
+            if not (a.is_hidden and user.is_in_first_year)
+        ]
```

Comment:
```
if user.is_in_first_year:
    queryset = queryset.filter(is_hidden=False)
```
would leave the filtering on the SQL side and be more efficient (especially for the pagination afterwards)

I'm not sure on the syntax here but you can check https://docs.djangoproject.com/en/4.0/ref/models/querysets/#filter